ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/ConnectIntegrationTest.java

```diff
@@ -266,18 +266,22 @@ public void shouldReadTimestampsFromConnect() {
         .put("topic.creation.default.partitions", "1")
         .build());
 
-    final long start = System.nanoTime();
-    RestResponse<KsqlEntityList> response;
-    do {
-      response = ksqlRestClient.makeKsqlRequest("CREATE STREAM FOO (PAYLOAD TIMESTAMP) WITH (KAFKA_TOPIC='foo', VALUE_FORMAT='DELIMITED');");
-    } while(!response.isSuccessful() && System.nanoTime() - start < TIMEOUT_NS);
+    makeKsqlRequest("CREATE STREAM TIMESTAMP_STREAM (PAYLOAD TIMESTAMP) WITH (KAFKA_TOPIC='foo', VALUE_FORMAT='DELIMITED');");
+    makeKsqlRequest("CREATE STREAM TIME_STREAM (PAYLOAD TIME) WITH (KAFKA_TOPIC='foo', VALUE_FORMAT='DELIMITED');");
+    makeKsqlRequest("CREATE STREAM DATE_STREAM (PAYLOAD DATE) WITH (KAFKA_TOPIC='foo', VALUE_FORMAT='DELIMITED');");
 
     // When:
-    final RestResponse<List<StreamedRow>> queryFoo = ksqlRestClient.makeQueryRequest("SELECT * FROM FOO EMIT CHANGES LIMIT 1;", 1L);
+    final RestResponse<List<StreamedRow>> queryTimestamp = ksqlRestClient.makeQueryRequest("SELECT * FROM TIMESTAMP_STREAM EMIT CHANGES LIMIT 1;", 1L);
+    final RestResponse<List<StreamedRow>> queryTime = ksqlRestClient.makeQueryRequest("SELECT CAST(PAYLOAD AS STRING) FROM TIME_STREAM EMIT CHANGES LIMIT 1;", 1L);
+    final RestResponse<List<StreamedRow>> queryDate = ksqlRestClient.makeQueryRequest("SELECT CAST(PAYLOAD AS STRING) FROM DATE_STREAM EMIT CHANGES LIMIT 1;", 1L);
```

Comment:
Shouldn't we wait on this PR until the string implicit casting is done automatically? Like the timestamp example.

Reply:
This was due to the REST API not converting Date/Time objects to String correctly. I ended up just adding that fix to this PR though, since it is a small fix.